File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
            unordered_map<int,int>gzip;
            for(int i=0;i<nums.size();i++){
                
                    if(gzip.find(target-nums[i])!=gzip.end()){
                                return {gzip[target-nums[i]],i};        
                    }
                    else{
                                gzip[nums[i]]=i;
                        
                    }
                
            }
        
            return {-1,-1};
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        d={}
        
        
        for i in range(len(nums)):
                
            val= target-nums[i]
            if val in d:
                
                return [d[val],i]
            
            else:
                
                d[nums[i]]=i
            
        
        return [-1,-1]
                